shellbase-core/src/main/scala/com/sumologic/shellbase/ShellBase.scala

```diff
@@ -163,7 +163,11 @@ abstract class ShellBase(val name: String)
     Thread.currentThread.setName("Shell main")
     try {
       val result = actualMain(args)
-      exitShell(result)
+      if (result != 0) {
+        exitShell(result)
```

Comment:
Does the program still execute from command line after executing successfully?  i.e. `myCustomerShell someCommand` from bash should exit still.

Reply:
Yeah. I've just double-checked that. With this code one of the shells we have have successfully run and completed its execution when executed from command line.
So no change in this use-case (as intended).